Andrii_Voinenko/Lesson12/task2.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+class Employee:
+    def __init__(self, name: str, start_day: str, salary: float, position: str):
+        self.__name = name
+        self.__start_day = start_day
+        self.__salary = salary
+        self.__position = position
+        self.__course_access = None
+
+        if self.__position not in ['junior', 'Junior', 'middle', 'Middle', 'senior', 'Senior']:
+            raise TypeError('Please set correct employee position')
+
+    @property
+    def name(self):
+        """
+        Return employee name
+        """
+        return self.__name
+
+    @property
+    def start_day(self):
+        """
+        Return date when employee was accepted
+        """
+        return datetime.strptime(self.__start_day, '%d.%m.%Y').date()
+
+    @property
+    def salary(self):
+        """
+        Return employee current salary
+        """
+        return self.__salary
+
+    @property
+    def position(self):
+        """
+        Return employee work position
+        """
+        return self.__position
+
+    @name.setter
+    def name(self, new_name):
+        """
+        Allows set new name to employee
+        """
+        self.__name = new_name
+
+    @salary.setter
+    def salary(self, new_salary):
+        """
+        Allows set new salary to employee
+        """
+        self.__salary = new_salary
```

Comment:
new salary can be any number. So it can be case when employee pays money to company